### PR TITLE
[IMP] estate: extend models and add UI for users T#69469

### PR DESCRIPTION
**Pull request by Suavesito-Olimpiada**
In this PR I modify the CRUD methods for `estate.property`. I also extend `res.users` model and views to improve UI/UX. The details are included in the commit message.

This is ready for review @deivislaya

Regards!

(I'll rebase over branch 16.0 once #12 is merged.)

### CODE REVIEW

**Comment by Suavesito-Olimpiada** [comment]
Rebased and ready for review @deivislaya.

File: estate/models/__init__.py [hunk 1]
```diff
@@ -4,3 +4,5 @@
 from . import estate_property_type
 from . import estate_property_tag
 from . import estate_property_offer
+
```
Comment (deivislaya) [inline estate/models/__init__.py]: Delete line

File: estate/models/estate_property_offer.py [hunk 2]
```diff
@@ -89,3 +89,16 @@ def action_refuse(self):
...
         return True
+
+    @api.model
+    def create(self, vals):
+        """ Creating a new offer sets the asociated property in the
+        offer_received state. If the property is sold or canceled, or if the
+        price for the new offer is less than the bigger offer, raise an error.
+        """
+        property_offered = self.env["estate.property"].browse(vals["property_id"])
+        message = _("Property cannot create new offers.")
```
Comment (deivislaya) [inline estate/models/estate_property_offer.py:100]: This message must be inside the `_check_availability` method. For modularity concept

File: estate/models/__init__.py (hunk 1 above)
Comment (deivislaya) [reply estate/models/__init__.py]: @Suavesito-Olimpiada 

Change pending

File: estate/models/estate_property_offer.py (hunk 2 above)
Comment (deivislaya) [reply estate/models/estate_property_offer.py:100]: @Suavesito-Olimpiada 
Change pending

File: estate/models/__init__.py (hunk 1 above)
Comment (Suavesito-Olimpiada) [reply estate/models/__init__.py]: Done.

File: estate/models/estate_property_offer.py (hunk 2 above)
Comment (Suavesito-Olimpiada) [reply estate/models/estate_property_offer.py:100]: This message is not the only one, in lines `74`, `87` and `101` there are similar messages, each different depending on the context from which `_check_availability` is called.

The context is the action being executed on the model `estate.property.offer`. Should, I instead pass the action and select the messages on `_check_availability` based on it?

Comment (Suavesito-Olimpiada) [reply estate/models/estate_property_offer.py:100]: `_check_availability` is also used in the `estate_property.py` file for the `estate.property` model.

I added a default error message.

**Comment by Suavesito-Olimpiada** [comment]
Finished and ready for review @deivislaya.

File: estate/models/estate_property.py [hunk 3]
```diff
@@ -135,11 +135,10 @@ def _onchange_garden(self):
...
-        UserError when the property is not available for sale, that is, if it
-        is sold or canceled. Receives a single record in `self` and an error
-        message.
+    def _check_availability(self, error_msg="Cannot complete the action."):
+        """ Errors if the property is not available. Raises an UserError when
+        the property is not available for sale, that is, if it is sold or
+        canceled. Receives a single record in `self` and an error message.
         """
         self.ensure_one()
         if self.state in ["sold", "canceled"]:
```
Comment (deivislaya) [inline estate/models/estate_property.py]:
```suggestion
    def _check_availability(self, error_msg=False):
        """ Errors if the property is not available. Raises an UserError when
        the property is not available for sale, that is, if it is sold or
        canceled. Receives a single record in `self` and an error message.
        """
        self.ensure_one()
        error_msg = error_msg or _("Cannot complete the action.")
        if self.state in ["sold", "canceled"]:
```

**Pull request merged**